picoquic/packet.c

```diff
@@ -50,221 +50,183 @@ int picoquic_parse_packet_header(
 
     /* Is this a long header of a short header? -- in any case, we need at least 17 bytes */
     if ((bytes[0] & 0x80) == 0x80) {
-        if (length < 17) {
+        if (length < 6) {
             ret = -1;
         } else {
-            /* If this is a long header, the bytes at position 9--12 describe the version */
-            ph->offset = 1 + picoquic_parse_connection_id(bytes + 1, 8, &ph->cnx_id);
-            ph->vn = PICOPARSE_32(bytes + ph->offset);
-            ph->offset += 4;
-
-            if (ph->vn == 0) {
-                /* VN = zero identifies a version negotiation packet */
-                ph->ptype = picoquic_packet_version_negotiation;
-                ph->pn = 0;
-                ph->pnmask = 0;
-                ph->pn_offset = 0;
-                ph->version_index = -1;
-
-                if (*pcnx == NULL) {
-                    *pcnx = picoquic_cnx_by_id(quic, ph->cnx_id);
+            uint8_t l_dest_id, l_srce_id;
+            /* The bytes at position 1..3 describe the version */
```

Comment:
The version is 4byte long, so shouldn't that byte 1-4?

Reply:
Yes.